Migrate debug page to hooks.

anuraaga opened the pull request:
It'd be nice if you can check out the debug page in this branch locally to make sure it works ok - I tried various operations and think it's fine but there are a lot of possible variations.

Also fixes long-time pet peeve by prettifying gRPC unframed JSON.

Fixes #1949

trustin commented on docs-client/src/lib/transports/grpc-unframed.ts:
```diff
@@ -51,7 +52,6 @@ export default class GrpcUnframedTransport extends Transport {
     });
-    const response = await httpResponse.text();
-    return response;
+    return jsonPrettify(await httpResponse.text());
```
:tada:

ikhoon commented:
While I was testing I found an exception. I clicked `Submit` button with any input and I saw a black page and exeception message.
```js

react-dom.development.js?61bb:13124 Uncaught TypeError: Method Error.prototype.toString called on incompatible receiver undefined
    at toString (<anonymous>)
    at basicStateReducer (react-dom.development.js?61bb:13124)
    at updateReducer (react-dom.development.js?61bb:13246)
    at updateState (react-dom.development.js?61bb:13294)
    at Object.useState (react-dom.development.js?61bb:13843)
    at useState (react.development.js?72d0:1462)
    at DebugPage (DebugPage.tsx?725f:115)
    at DebugPage (react-hot-loader.development.js?c2cb:821)
    at renderWithHooks (react-dom.development.js?61bb:12938)
    at updateFunctionComponent (react-dom.development.js?61bb:14627)
```
![TestSrevice UnaryCall2](https://user-images.githubusercontent.com/1866157/66300094-ac99b500-e92f-11e9-87b3-dd681975340e.gif)

anuraaga commented:
@ikhoon Thanks for finding, very silly bug!

ikhoon commented:
@anuraaga Your PR is always welcome. I'm happy to find and share it 😉

ikhoon commented:
I think `404 Bad Request` message doesn't print on debug console.

![unary-method](https://user-images.githubusercontent.com/1866157/66534029-f19c3200-eb4f-11e9-8385-b404b6fda3b8.gif)

This is master version screen shot.😀
<img width="1402" alt="Screen Shot 2019-10-10 at 11 13 54 AM" src="https://user-images.githubusercontent.com/1866157/66533957-b699fe80-eb4f-11e9-8ea6-b772daadeb58.png">

anuraaga commented:
Thanks - fixed

anuraaga commented:
I think both were introduced in previous PRs (by me ><). Fixed the error, and also removed a spacing parameter that was causing the hover issue. The method type / name are a bit closer now, but I guess it still looks ok to me, though I might be forgetting the reason the parameter was there in the first place.
